pkg/controllers/edgeconnect/controller.go

```diff
@@ -676,32 +714,93 @@ func (controller *Controller) createOrUpdateEdgeConnectDeployment(ctx context.Co
 		return err
 	}
 
+	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
```

Comment:
`createOrUpdateEdgeConnectDeployment` function is not related to Deployment any more.

maybe `createOrUpdateEdgeConnectDeploymentAndSettings` name would be better or a new wrapper function could be added to group Deployment and Settings changes together and to keep `createOrUpdateEdgeConnectDeployment` function focused on a single task

Reply:
Agree with you :+1: 
Updated [here](https://github.com/Dynatrace/dynatrace-operator/pull/3206/commits/8628226038be2c03a62e1b3d9ed753143213ddcf)